tslearn/barycenters/dba.py

```diff
@@ -253,19 +263,22 @@ def _subgradient_valence_warping(list_p_k, barycenter_size, weights):
        for Averaging in Dynamic Time Warping Spaces.
        Pattern Recognition, 74, 340-358.
     """
-    list_v_k = []
-    list_w_k = []
-    for k, p_k in enumerate(list_p_k):
-        sz_k = p_k[-1][1] + 1
-        w_k = numpy.zeros((barycenter_size, sz_k))
+    def create_w_k(p_k, barycenter_size):
+        w_k = numpy.zeros((barycenter_size, p_k[-1][1] + 1))
         for i, j in p_k:
             w_k[i, j] = 1.
-        list_w_k.append(w_k * weights[k])
-        list_v_k.append(w_k.sum(axis=1) * weights[k])
+        return w_k
+    w_k_ones = Parallel(backend='threading',prefer='threads',require='sharedmem', n_jobs=n_treads, verbose = True)(delayed(create_w_k)(p_k,barycenter_size) for p_k in list_p_k)
```

Comment:
Is this part really crucial to parallelize? I would expect that it would be neglectible in terms of running times.

Reply:
This function cannot fully utilize CPU cores, even if we force to parallelize dba on different clusters and then becomes the bottleneck.